_posts/2021-02-01-aws-config-compliance-automation.md

```diff
@@ -0,0 +1,95 @@
+
+## ServiceNow
+
+In large companies, many choose to utilize workflow automation software, and one of the most popular of these systems, is the cloud-based <a href="https://www.servicenow.com/">ServiceNow</a> platform. ServiceNow can be utilized in a variety of ways, but normally it is used as the source-of-truth for internal configuration management database (CMDB) needs, allowing an organization to have visibility and centralized control of disperse assets. Another use of ServiceNow is to help manage IT processes around internal Help Desk, as well as problem and incident tracking. These types of requests generally blend together with CMDB needs to the point where a central tool encompassing all of these needs is very beneficial.
+
+When considering the normal IT process workflows that some companies help to automate with the help of ServiceNow, compliance needs become very similar. The normal mode of operation for IT assets would be to have them in compliance with regulations and policy, so when an asset is found to be non-compliant, this is an issue just like several other IT issues (service outages, request tracking, other incicents, etc), and as such it makes sense to track utilizing the same system as those other issues.
+
+If ServiceNow is accurately reflecting configuration item (CI) data in the CMDB, there are some useful direct integrations possible using a combination of AWS Config, AWS Simple Notification Service (SNS), and ServiceNow. However, many organizations do not have a 1:1 reflection of AWS resources as CI's in their ServiceNow CMDB, so some customization is required. Unfortunately when this is necessary, the <a href="https://docs.servicenow.com/bundle/paris-it-operations-management/page/product/cloud-management-v2-setup/task/aws-config-service-cloud-mgt.html">out-of-the-box capabilities utilizing the tools</a> is not possible.
+
+## Alternative solution
+
+In these cases, some customization is necessary to drive visibility of the non-compliant items from AWS Config into the normal resolution workflow in ServiceNow. A solution that is possible, is to pull data from AWS Config using `aws-sdk` API's, and then feeding that information into ServiceNow using ServiceNow API's.
+
+Many organizations deploy resources into several AWS accounts, making management of these items a little trickier. Luckily, AWS Config provides a capability to track compliance data at an <a href="https://docs.aws.amazon.com/config/latest/developerguide/aggregate-data.html">aggregate-level, across many accounts</a>. Once aggregated, the data is made available in a few different ways, but the most flexible seems to be utilizing <a href="https://docs.aws.amazon.com/config/latest/developerguide/querying-AWS-resources.html">AWS Config Advanced Queries</a> to pull back compliance information across many accounts while utilizing the same method to get the data for each. 
```

Comment:
"Once aggregated, the data is made available in a few different ways. The most flexible of which seems..."